azdev/operations/secret.py

```diff
@@ -0,0 +1,242 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -----------------------------------------------------------------------------
+
+import os
+import json
+from knack.log import get_logger
+from microsoft_security_utilities_secret_masker import (load_regex_patterns_from_json_file,
+                                                        load_regex_pattern_from_json,
+                                                        SecretMasker)
+logger = get_logger(__name__)
+
+
+def _validate_data_path(file_path=None, directory_path=None, data=None):
+    if file_path and directory_path:
+        raise ValueError('Can not specify file path and directory path at the same time')
+    if file_path and data:
+        raise ValueError('Can not specify file path and raw string at the same time')
+    if directory_path and data:
+        raise ValueError('Can not specify directory path and raw string at the same time')
+    if not file_path and not directory_path and not data:
+        raise ValueError('No file path ot directory path or raw string provided')
+
+    if directory_path and not os.path.isdir(directory_path):
+        raise ValueError(f'invalid directory path:{directory_path}')
+    if file_path and not os.path.isfile(file_path):
+        raise ValueError(f'invalid file path:{file_path}')
+
+
+def _load_built_in_regex_patterns():
+    return load_regex_patterns_from_json_file('PreciselyClassifiedSecurityKeys.json')
+
+
+def _load_regex_patterns(custom_pattern=None):
+    built_in_regex_patterns = _load_built_in_regex_patterns()
+
+    if not custom_pattern:
+        return built_in_regex_patterns
+    if os.path.isfile(custom_pattern):
+        custom_pattern = json.load(custom_pattern)
```

Comment:
Does custom_pattern have to be in JSON format? Do we need to provide additional error handling, such as capturing json.JSONDecodeError and providing a more friendly error prompt?

Reply:
added